torch/fx/node.py

```diff
@@ -64,22 +78,23 @@ def replace_all_uses_with(self, replace_with : 'Node') -> List['Node']:
         Replace all uses of `self` in the Graph with the Node `replace_with`.
         Returns the list of nodes on which this change was made.
         """
-        use_nodes : List[Node] = self.find_uses()
-        for use_node in use_nodes:
+        to_process = list(self.users)
+        for use_node in to_process:
             def maybe_replace_node(n : Node) -> Node:
                 if n == self:
-                    self.uses -= 1
                     return replace_with
                 else:
                     return n
+
             new_args = map_arg(use_node.args, maybe_replace_node)
             assert isinstance(new_args, tuple)
             use_node.args = new_args
             new_kwargs = map_arg(use_node.kwargs, maybe_replace_node)
             assert isinstance(new_kwargs, dict)
             use_node.kwargs = new_kwargs
 
-        return use_nodes
+        self.users.clear()
```

Comment:
with the current code this should be an assert right? because args/kwargs updates should have done it. Alternative that is more efficient would be to directly assign to `_args/_kwargs` above and then call clear here.